windows/readiness_healthcheck.go

```diff
@@ -0,0 +1,98 @@
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/skip_messages"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = WindowsDescribe("Readiness Healthcheck", func() {
+	var appName string
+	var readinessHealthCheckTimeout = "25s" // 20s route emitter sync loop + 2s hc interval + bonus
+
+	BeforeEach(func() {
+		if !Config.GetReadinessHealthChecksEnabled() {
+			Skip(skip_messages.SkipReadinessHealthChecksMessage)
+		}
+		appName = random_name.CATSRandomName("APP")
+	})
+
+	AfterEach(func() {
+		app_helpers.AppReport(appName)
+		Eventually(cf.Cf("delete", appName, "-f")).Should(Exit(0))
+	})
+
+	// TODO add a ready endpoint to Nora
+	Describe("when the readiness healthcheck is set to http", func() {
+		It("registers the route only when the readiness check passes", func() {
+			By("pushing the app")
+			// FIXME use manifest and update linux
+			Expect(cf.Cf("push",
+				appName,
+				"-f", assets.NewAssets().Nora+"/../readiness_manifest.yml",
+				"-p", assets.NewAssets().Nora,
+			).Wait(Config.CfPushTimeoutDuration())).To(Exit(0))
+
+			By("verifying the app starts")
+			Eventually(func() string {
+				return helpers.CurlAppRoot(Config, appName)
+			}, Config.CfPushTimeoutDuration()).Should(ContainSubstring("hello i am nora running on"))
+
+			By("verifying the app is marked as ready")
+			// TODO: only include this when audit events are built
+			// Eventually(func() string {
+			// 	return string(cf.Cf("events", appName).Wait().Out.Contents())
+			// }).Should(MatchRegexp("app.ready"))
+
+			// Windows apps take a Loooooong time to start apparently
+			Expect(string(logs.Recent(appName).Wait(Config.CfPushTimeoutDuration()).Out.Contents())).Should(ContainSubstring("Container passed the readiness health check"))
+
+			By("triggering the app to make the /ready endpoint fail")
+			helpers.CurlApp(Config, appName, "/ready/false")
+
+			By("verifying the app is marked as not ready")
+
+			// TODO: only include this when audit events are built
+			// Eventually(func() string {
+			// 	return string(cf.Cf("events", appName).Wait().Out.Contents())
+			// }).Should(MatchRegexp("app.notready"))
+
+			Eventually(func() string {
+				return string(logs.Recent(appName).Wait().Out.Contents())
+			}, readinessHealthCheckTimeout).Should(ContainSubstring("Container failed the readiness health check"))
+
+			By("verifying the app is removed from the routing table")
+			Eventually(func() string {
+				return helpers.CurlApp(Config, appName, "/ready")
+			}).Should(ContainSubstring("404 Not Found"))
+
+			By("verifying that the app hasn't restarted")
+			Consistently(func() string {
+				return string(cf.Cf("events", appName).Wait().Out.Contents())
+			}).ShouldNot(MatchRegexp("audit.app.process.crash"))
+
+			By("re-enabling the app's readiness endpoint")
+			Expect(cf.Cf("ssh", appName, "-c", "curl localhost:8080/ready/true").Wait(Config.DefaultTimeoutDuration())).To(Exit(0))
```

Comment:
Is there another way to re-enable the app's readiness endpoint? The Diego container ssh feature is not guaranteed to be turned on for this suite.

Reply:
Unfortunately no. The method used for linux is to use c2c networking (which isnt' eaven a feature on windows). 
we can remove the re-enable checks if it sounds like this will be a large concern.

Reply:
or maybe there's a boolean for app ssh we can hide it behind?

Reply:
Hmmmm ok in that case let's roll with this for now. In the worst-case, I can fast-follow this major release with a patch to gate this test on the `include_ssh` config option.